pkg/metrics/metrics.go

```diff
@@ -173,13 +190,53 @@ func (m *Metrics) UpdateCertificateExpiry(crt *v1alpha1.Certificate, secretListe
 		return
 	}
 
-	updateX509Expiry(crt.Name, crt.Namespace, cert)
+	updateX509Expiry(crt, cert)
 }
 
-func updateX509Expiry(name, namespace string, cert *x509.Certificate) {
+func updateX509Expiry(crt *v1alpha1.Certificate, cert *x509.Certificate) {
 	// set certificate expiry time
 	expiryTime := cert.NotAfter
+	RegisteredCertificates.mtx.Lock()
+	defer RegisteredCertificates.mtx.Unlock()
 	CertificateExpiryTimeSeconds.With(prometheus.Labels{
-		"name":      name,
-		"namespace": namespace}).Set(float64(expiryTime.Unix()))
+		"name":      crt.Name,
+		"namespace": crt.Namespace}).Set(float64(expiryTime.Unix()))
+	RegisteredCertificates.certificates[crt] = struct{}{}
+}
+
+func (m *Metrics) CleanUp(certificateLister cmlisters.CertificateLister) {
+	log := logf.FromContext(m.ctx)
+	log.V(logf.DebugLevel).Info("attempting to clean up deleted certificates")
```

Comment:
`attempting to clean up metrics for recently deleted certificates` makes it a bit clearer I think 😄